examples/multi-decade-example.py

```diff
@@ -32,16 +32,17 @@
 
 # when a pypsa network from an older version is imported, the old snapshot weightings
 # are set row-wise and a warning is given
-n = pypsa.Network("opf-storage-hvdc/opf-storage-data")
+n = pypsa.examples.storage_hvdc()
 print(n.snapshot_weightings)
 
 # snapshots can be set as a multiindex
-# CAUTION time-dependent data is lost from changing single to multiindex
+# a convenience functionality is provided: when setting the investment periods
+# directly, the all time-series are repeated for each period.
 single_snapshots = n.snapshots
-snapshots_multiindex = pd.MultiIndex.from_product([[2020, 2030], single_snapshots])
-n.set_snapshots(snapshots_multiindex)
+n.investment_periods = [2020, 2030]
 print(n.snapshots)
 print(n.investment_period_weightings)
+assert all(n.snapshots == pd.MultiIndex.from_product([[2020, 2030], single_snapshots]))
```

Comment:
The subsequent test with non-ascending investment periods no longer generates an error message
`snapshots_multiindex = pd.MultiIndex.from_product([[2030, 2020], single_snapshots])
 n.set_snapshots(snapshots_multiindex)`
It does check when setting the investment periods that they are integer and increasing and if all investment periods are defined in the network snapshots, but it would be good if it also checks the order of the first level index of the snapshots. Otherwise it is possible to have like in this case snapshots with level=0 order [2030,2020] and investment periods with order [2020,2030]